packages/eslint-plugin/src/rules/unit-naming-rule.ts

```diff
@@ -1,265 +1,228 @@
-      })
-      return
-    }
-    if (!unitIdentifier.name.endsWith(atomPostfix)) {
-      context.report({
-        node: nameArg,
-        messageId: 'postfixMissing',
-        fix: (fixer) => {
-          const x = insertUnitName({
-            selfName: unitIdentifier.name + atomPostfix,
-            isPrivate,
-            domain,
-          })
-          return [
-            fixer.replaceText(nameArg, x),
-            fixer.replaceText(
-              unitIdentifier,
-              unitIdentifier.name + atomPostfix,
-            ),
-          ]
-        },
-      })
-      return
-    }
-  }
-}
-
-const insertUnitName = ({
-  selfName,
-  domain,
-  isPrivate = false,
-}: {
-  selfName: string
-  domain: AutoDomain | string | null
-  isPrivate?: boolean
-}) => {
-  if (isPrivate) selfName = '_' + selfName
-  if (domain === AutoDomain) return '`${name}.' + selfName + '`'
-  if (typeof domain === 'string') return `'${domain}.${selfName}'`
-  return `'${selfName}'`
-}
+import * as estree from 'estree'
+import { Rule } from 'eslint'
+import { patternNames, reatomFactoryPattern } from '../shared'
+
+type Domain = { is: 'absent' } | { is: 'static'; name: string } | { is: 'dynamic'; vary: string }
+
+interface Name {
+  domain: Domain
+  local: boolean
+  object: string | null
+  unit: string
+}
+
+export const unitNamingRule: Rule.RuleModule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      recommended: true,
+      description: 'Ensures that all Reatom entities are correctly named.',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          atomPrefix: {
+            type: 'string',
+          },
+          atomPostfix: {
+            type: 'string',
+          },
+          domainVariable: {
+            type: 'string',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    fixable: 'code',
+  },
+
+  create(context) {
+    const {
+      atomPrefix = '',
+      atomPostfix = '',
+      domainVariable = 'name',
+    }: {
+      atomPrefix?: string
+      atomPostfix?: string
+      domainVariable?: string
+    } = context.options[0] ?? {}
+
+    const domainScopes = [] as (Domain | null)[]
+    const idScopes = [] as (estree.Identifier | null)[]
+
+    return {
+      ':matches(VariableDeclarator, Property)'(node: estree.VariableDeclarator | estree.Property) {
+        const id = node.type === 'Property' ? node.key : node.id
+        idScopes.push(id.type === 'Identifier' ? id : null)
+      },
+      ':matches(VariableDeclarator, Property):exit'() {
+        idScopes.pop()
+      },
+      [`:function`](node: estree.Function) {
+        const declarators =
+          node.body.type === 'BlockStatement'
+            ? node.body.body.flatMap((statement) => {
+                return statement.type === 'VariableDeclaration' ? statement.declarations : []
+              })
+            : []
+        const patterns = [...node.params, ...declarators.map((d) => d.id)]

```

Comment:
```suggestion
        const declarations =
          node.body.type === 'BlockStatement'
            ? node.body.body.flatMap((statement) => {
                return statement.type === 'VariableDeclaration' ? statement.declarations : []
              })
            : []
        const patterns = [...node.params, ...declarations.map((d) => d.id)]

```

As we return `statement.declarations` makes more sense for me to call variable `declarations`, wdyt?

Reply:
https://github.com/estree/estree/blob/master/es5.md#variabledeclaration